doc/filters/merge.rst

```diff
@@ -40,6 +40,20 @@ overridden.
 
         {# items now contains { 'apple': 'fruit', 'orange': 'fruit' } #}
         
+.. tip::
+
+    If the values or keys you want to merge are variable, you need to enclose them into parenthesis :
+
+    .. code-block:: jinja
+        {% set mergedItems = {} %}
+        {% set items = { 'apple': 'fruit', 'orange': 'fruit' } %}
+
+        {% for key, value in items %} 
+            {% set mergedItems = mergedItems|merge({ (key): (value) }) %}
```

Comment:
Ok. Care on a note about the array_merge particular behavior with numeric variable as key ?